shm/Map.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "Island.hpp"
+#include <vector>
+
+constexpr size_t amountOfIslands = 10;
+constexpr size_t minPositionXY = 0;
+constexpr size_t maxPositionXY = 10;
+
+class Map {
+private:
```

Comment:
in some of your classes public is first, sometimes private. If you want you can use them is the same order every time, it will be a bit easier to read and manage

Reply:
+1